pkg/conversions/virtualrouter_types_conversion.go

```diff
@@ -0,0 +1,327 @@
+	if len(crdObj.Headers) != 0 {
+		sdkHeaders = make([]*appmeshsdk.HttpRouteHeader, 0, len(crdObj.Headers))
+		for _, crdHeader := range crdObj.Headers {
+			sdkHeader := &appmeshsdk.HttpRouteHeader{}
+			if err := Convert_CRD_HTTPRouteHeader_To_SDK_HttpRouteHeader(&crdHeader, sdkHeader, scope); err != nil {
+				return err
+			}
+			sdkHeaders = append(sdkHeaders, sdkHeader)
+		}
+	}
+
+	sdkObj.Headers = sdkHeaders
+	sdkObj.Method = crdObj.Method
+	sdkObj.Prefix = aws.String(crdObj.Prefix)
+	sdkObj.Scheme = crdObj.Scheme
+	return nil
+
+}
+
+func Convert_CRD_HTTPRouteAction_To_SDK_HttpRouteAction(crdObj *appmesh.HTTPRouteAction,
+	sdkObj *appmeshsdk.HttpRouteAction, scope conversion.Scope) error {
+
+	var sdkWeightedTargets []*appmeshsdk.WeightedTarget
+	if len(crdObj.WeightedTargets) != 0 {
+		sdkWeightedTargets = make([]*appmeshsdk.WeightedTarget, 0, len(crdObj.WeightedTargets))
+		for _, crdWeightedTarget := range crdObj.WeightedTargets {
+			sdkWeightedTarget := &appmeshsdk.WeightedTarget{}
+			if err := Convert_CRD_WeightedTarget_To_SDK_WeightedTarget(&crdWeightedTarget, sdkWeightedTarget, scope); err != nil {
+				return err
+			}
+			sdkWeightedTargets = append(sdkWeightedTargets, sdkWeightedTarget)
+		}
+	}
+	sdkObj.WeightedTargets = sdkWeightedTargets
+	return nil
+}
+
+func Convert_CRD_HTTPRetryPolicy_To_SDK_HttpRetryPolicy(crdObj *appmesh.HTTPRetryPolicy,
+	sdkObj *appmeshsdk.HttpRetryPolicy, scope conversion.Scope) error {
+
+	var sdkHttpRetryEvents []*string
+	if len(crdObj.HTTPRetryEvents) != 0 {
+		sdkHttpRetryEvents = make([]*string, 0, len(crdObj.HTTPRetryEvents))
+		for _, crdHTTPRetryEvent := range crdObj.HTTPRetryEvents {
+			sdkHttpRetryEvents = append(sdkHttpRetryEvents, aws.String((string)(crdHTTPRetryEvent)))
+		}
+	}
+	sdkObj.HttpRetryEvents = sdkHttpRetryEvents
+
+	var sdkTcpRetryEvents []*string
+	if len(crdObj.TCPRetryEvents) != 0 {
+		sdkTcpRetryEvents = make([]*string, 0, len(crdObj.TCPRetryEvents))
+		for _, crdTCPRetryEvent := range crdObj.TCPRetryEvents {
+			sdkTcpRetryEvents = append(sdkTcpRetryEvents, aws.String((string)(crdTCPRetryEvent)))
+		}
+	}
+	sdkObj.TcpRetryEvents = sdkTcpRetryEvents
+
+	sdkObj.PerRetryTimeout = &appmeshsdk.Duration{}
+	if err := Convert_CRD_Duration_To_SDK_Duration(&crdObj.PerRetryTimeout, sdkObj.PerRetryTimeout, scope); err != nil {
+		return err
+	}
+
+	sdkObj.MaxRetries = aws.Int64((int64)(crdObj.MaxRetries))
+	return nil
+}
+
+func Convert_CRD_HTTPRoute_To_SDK_HttpRoute(crdObj *appmesh.HTTPRoute,
+	sdkObj *appmeshsdk.HttpRoute, scope conversion.Scope) error {
+
+	sdkObj.Match = &appmeshsdk.HttpRouteMatch{}
+	if err := Convert_CRD_HTTPRouteMatch_To_SDK_HttpRouteMatch(&crdObj.Match, sdkObj.Match, scope); err != nil {
+		return err
+	}
+
+	sdkObj.Action = &appmeshsdk.HttpRouteAction{}
+	if err := Convert_CRD_HTTPRouteAction_To_SDK_HttpRouteAction(&crdObj.Action, sdkObj.Action, scope); err != nil {
+		return err
+	}
+
+	sdkObj.RetryPolicy = &appmeshsdk.HttpRetryPolicy{}
```

Comment:
we need to check crdObj. RetryPolicy nil here

Reply:
Done